log.go

```diff
@@ -417,6 +419,63 @@ func (l *subLog) Criticalf(format string, args ...any) {
 	}
 }
 
+// TraceS writes a structured log with the given message and key-value pair
+// attributes with LevelTrace to the log.
+//
+// This is part of the Logger interface implementation.
+func (l *subLog) TraceS(_ context.Context, msg string, attrs ...any) {
+	l.Tracef(msg, attrs...)
```

Comment:
Do we need to process the `attrs` first? Or is the assumption that `msg` would never contain formatting directives? Then perhaps we should call into `l.Trace()` instead.
Example of what I mean:
```
bar := "bar"
subLog.TraceS(ctx, "incorrectly used directive: %v", "foo", bar)
```

What I think would be printed:
```
incorrectly used directive: foo%!(EXTRA string=bar)
```

What the user would expect to be printed:
```
incorrectly used directive: foo=bar
```

Reply:
cool yeah agreed

Reply:
ah - the thing is thought that for this implementation, under the hood there is not actually structured logging going on. So the output we get will not look like "key=value" etc. So I used the `*f` methods just so that we are passing the `msg` and `attr` to some underlying impl. the non `f` methods only take `...attr` so the message would not be used. 

So this was just to somewhat complete this implementation. In reality if callers want structured logging, they should use the new slog implementation. In the LL fork PR, I completely remove this impl anyways. So I think instead of trying to get this to work as if it is using slog underneath, let's just add a warning to the impl saying it works like the `*f` methods. 

Taking a step back: i think we arent gonna get this into this repo anyways and so we will carry on with the forked version which removes this impl anyways

Reply:
EDIT given latest updates to the convo: perhaps I should just look into a best effort way of converting the given attributes to k=v pairs and writing those to the writer